x12-edi/2.6/modules/ROOT/pages/x12-edi-schema-language-reference.adoc

```diff
@@ -0,0 +1,385 @@
+|items |List of segments and potentially nested groups that comprise the group.
+|===
+
+== Segment Definitions
+
+Segment definitions are comprised of certain key-value pairs for standard characteristics along with lists of values such as elements and composites that make up the actual segment. The following example shows a portion of a sample segment definition, from the `/x12/005010/basedefs.esl` schema
+
+`M` is mandatory
+`O` is Optional
+
+[source,yaml,linenums]
+----
+- id: 'BAK'
+  name: Beginning Segment for Purchase Order Acknowledgment
+  values:
+  - { idRef: '353', usage: M }
+  - { idRef: '587', usage: M }
+  - { idRef: '324', usage: M }
+  - { idRef: '373', usage: M }
+  - { idRef: '328', usage: O }
+  - { idRef: '326', usage: O }
+----
+
+Segment definition values are:
+
+[%header, cols="20a,80a"]
+|===
+|Section |Description
+|id |segment identifier
+|name |segment name
+|values |list of elements and composites within the segment
+|===
+
+The values list references elements and composites by ID, and use a compact YAML syntax: the values for each are shown as comma-separated key-value pairs enclosed in curly braces.
+
+[%header, cols="20a,80a"]
+|===
+|Section |Description
+|idRef |The referenced element or composite ID.
+|position |The value position within the segment starts at 1 and increases by 1 for each successive value. Generally not used.
+|name |The name of the value in the segment, by default the element or composite name is used.
+|usage |Usage code definition:
+
+* `C` for Conditional
+* `M` for Mandatory
+* `O` for Optional
+* `U` for Unused, the item should not appear at all in the data.
+* `I` for Ignored, this can be present in data, as in the case of the HL segments it must be present, but is not visible for the user in metadata.
+
+|count |Maximum repetition count value that can be a number or the special value >1 . This means any number of repeats. A count value of 1 is used if not specified.
+|===
+
+== Composite Definitions
+
+Composite definitions are very similar to segment definitions. They are composed of certain key-value pairs for standard characteristics along with lists of values  such as elements and composites that make up the actual composite. This example shows a portion of a composite definition, from the `/x12/005010/basedefs.esl schema`.
+
+[source,yaml,linenums]
+----
+- id: 'C022'
+  name: 'Health Care Code Information'
+  values:
+  - { idRef: '1270', usage: M }
+  - { idRef: '1271', usage: M }
+  - { idRef: '1250', usage: C }
+  - { idRef: '1251', usage: C }
+  - { idRef: '782', usage: O }
+----
+
+Composite definition values are:
+
+[%header, cols="20a,80a"]
+|===
+|Name |Description
+|id |composite identifier
+|name |composite name
+|values |list of elements and composites within the composite
+|===
+
+The values list references elements and composites by ID, and use a compact YAML syntax: the values for each are shown as comma-separated key-value pairs enclosed in curly braces.
```

Comment:
I'm not sure about using a colon that way. I commented previously, but disregard if you think it's fine. And it's really not important right now.